### PR TITLE
Add a Label widget

### PR DESCRIPTION
**Pull request by davep**
For the moment this does nothing more than inherit from a Static; but what it does do is make it easier for someone to add text to their application and to style it by styling all the Labels. Before now it would be common to use a Static but if you try and style (or query) all Statics, you'd also get things like Buttons, which inherit from Static.

See #1190

### CODE REVIEW

File: src/textual/widgets/_label.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"""Provides a simple Label widget."""
+
+from ._static import Static
+
+
+class Label(Static):
+    """A simple label widget for displaying text-oriented rendenrables."""
```
Comment (rodrigogiraoserrao) [inline src/textual/widgets/_label.py:7]:
```suggestion
    """A simple label widget for displaying text-oriented renderables"""
```

**Comment by UriyaHarpeness** [comment]
This feels weird...
I get the idea but if a user would want to change the style of only part of them he'd still need to do work.
And if going with this approach, I think a warning message should be added if constructing a Static directly and not through inheriting it.
What do you think @davep?

**Comment by davep** [comment]
> I get the idea but if a user would want to change the style of only part of them he'd still need to do work.

Obviously, hence classes and styles and the like in CSS. This early split between a more "baseline" renderable container and a more "has a specific job" renderable container isn't about reducing or eliminating that sort of work (which, as we both know, would be very little anyway), it just starts to set up what seems like a fairly natural split between a basic renderable container that's about showing text, and more involved widgets.

It's possible that at some point in the near future `Label` may acquire some extra functionality that wouldn't make much sense in `Static` but which anyone using `Label` would optionally want. This wee addition acknowledges that.

> And if going with this approach, I think a warning message should be added if constructing a Static directly and not through inheriting it.

I'm not really seeing much value in emitting warnings for this. As it stands, using `Static` rather than `Label` makes zero difference and has zero cost. What we're likely to do though is update the documentation and example code to naturally show that labels in an application will use `Label`, buttons will use `Button`, etc... and `Static` will sort of transitional into being a more foundational type.

File: docs/widgets/label.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Label
+
+A widget which displays static text, but which can also contain more complex Rich renderables.
+
+- [ ] Focusable
+- [x] Container
```
Comment (willmcgugan) [inline docs/widgets/label.md:6]: A Label can't contain children, which means it isn't a container. I think this is inherited from Static, which is also incorrect.

**Review by willmcgugan — APPROVED** [review]
Just one request for a doc tweak.

File: docs/widgets/label.md (hunk 2 above)
Comment (davep) [reply docs/widgets/label.md:6]: Good catch, I'll fix `Static` and then sync this.

Comment (davep) [reply docs/widgets/label.md]: > A Label can't contain children, which means it isn't a container. I think this is inherited from Static, which is also incorrect.

Actually... this isn't this isn't quite true is it? Especially not for static. A Static can comprise of other widgets. Consider [the stopwatch tutorial](https://textual.textualize.io/tutorial/#custom-widgets) for example.

**Pull request merged**